packages/backend/src/api/routers/FrontendRouter.ts

```diff
@@ -1,95 +1,33 @@
 import { PedersenHash } from '@explorer/crypto'
-import {
-  HomeProps,
-  renderHomePage,
-  renderPositionDetailsPage,
-  renderStateChangeDetailsPage,
-  renderStateChangesIndexPage,
-} from '@explorer/frontend'
 import Router from '@koa/router'
 
-import { StateUpdateRepository } from '../../peripherals/database/StateUpdateRepository'
+import { FrontendController } from '../controllers/FrontendController'
 
-export function createFrontendRouter(
-  stateUpdateRepository: StateUpdateRepository
-) {
+export function createFrontendRouter(frontendController: FrontendController) {
   const router = new Router()
 
   router.get('/', async (ctx) => {
-    const stateUpdates = await stateUpdateRepository.getStateChangeList({
-      offset: 0,
-      limit: 20,
-    })
-
-    ctx.body = renderHomePage({
-      forcedTransaction: [],
-      stateUpdates: stateUpdates.map(
-        (x): HomeProps['stateUpdates'][number] => ({
-          hash: x.rootHash,
-          timestamp: x.timestamp,
-          positionCount: x.positionCount,
-        })
-      ),
-    })
+    ctx.body = await frontendController.getHomePage()
   })
 
   router.get('/state-updates', async (ctx) => {
     const page = parseInt(String(ctx.query.page ?? '1'))
     const perPage = parseInt(String(ctx.query.perPage ?? '10'))
-
-    const stateUpdates = await stateUpdateRepository.getStateChangeList({
-      offset: (page - 1) * perPage,
-      limit: perPage,
-    })
-    const fullCount = await stateUpdateRepository.getStateChangeCount()
-
-    ctx.body = renderStateChangesIndexPage({
-      stateUpdates: stateUpdates.map((update) => ({
-        hash: update.rootHash,
-        timestamp: update.timestamp,
-        positionCount: update.positionCount,
-      })),
-      fullCount: Number(fullCount),
-      params: {
-        page,
-        perPage,
-      },
-    })
+    if ([page, perPage].some(Number.isNaN)) {
```

Comment:
We'd also want to change this logic so that we parse `ctx.query` using zod. This can be implemented as part of another PR.

Reply:
These were my thoughts as well. However, the router which is responsible for request data validation and parsing was not doing that for this specific route (unlike other routes where `PedersenHash` and `BigInt` constructors were handling it), so I added it to stay consistent. Otherwise we rely on the controller.